✨ Feat: Jwt 기능 구현

JeekLee opened the pull request:
## PR 체크사항
PR이 다음 사항을 만족하는지 확인해주세요.

<!-- 
체크하려면 괄호 안에 "x"를 입력하세요. 
각 규칙은 Convention 문서에 있습니다.
PR 제목에 쓰는 prefix는 다음과 같습니다.
🚀 Release
🐛 Fix
✨ Feat
📝 Doc
♻️ Refactor
🔧 Chore
⏪️ Revert
🧪 Test
🎉 Init
-->

- [x] 커밋 제목 규칙
- [x] 커밋 메시지 작성 가이드라인
- [x] 라벨, 담당자, 리뷰어 지정


## PR 타입
어떤 유형의 PR인지 체크해주세요.

<!-- 체크하려면 괄호 안에 "x"를 입력하세요. -->

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:


## PR 설명
- AccessToken/RefreshToken 방식으로 구현.
- 기능 구현을 위해 AccessToken 30분, RefreshToken 1시간으로 설정
- 현재 URI에 따라 인증/인가를 벗어날 수 있도록 설정해둔 상태입니다.
- 현재 API 구조 상, 같은 URI에 CURD가 구성되어 있는 경우가 있어 제대로된 적용이 어렵습니다.
- 때문에 일부 API들을 <span style="color:blue">**토큰 인증이 필요하지만 인증 없이도 작성이 되는 경우**</span>가 있을 수 있습니다.
- Http Method에 따라 예외를 설정하는 방법을 고민해보고 있는데, 쉽지 않아 우선 PR합니다.
- 위에 말씀드린 문제를 해결할 수 있는 방법이 있다면 알려주시면 너무 감사할 것 같고, 추가적으로 더 찾아보겠습니다!

## 작업사항
- JwtAuthFilter 구현
- JwtUtil 구현
- JwtExceptionFilter 구현
- CustomSecurityException 구현
- WebSecurityConfig 수정 (Jwt 적용)
- UserService 수정 (Jwt 적용)
- Response 수정 (@Getter가 없어 정상적으로 Response 객체가 생성되지 않는 문제 해결)


## 변경로직
- 내용을 적어주세요.
close #26

JeekLee committed:
feat: Jwt 사용을 위한 의존성 추가

JeekLee committed:
fix: Response 객체 fix

@Getter가 없어 정상적으로 Response 반환이 안되는 문제 해결

JeekLee committed:
feat: Token 관련 Exception 메세지 추가

1. 토큰 없음
2. 유효하지 않은 토큰
3. 로그아웃된 사용자(RefreshToken 없음)
4. 인가되지 않은 사용자

JeekLee committed:
feat: Jwt 관련 기능 및 필터 추가

1. JwtUtil
전반적인 기능을 구현
2. JwtAuthFilter
Filter에 Jwt 관련 조건 적용

JeekLee committed:
feat: Jwt Exception Handling을 위한 기능 추가

1. CustomSecurityException
기본적으로 RuntimeException을 상속하며, Filter 단에서 발생한
Exception을 처리하기 위해 사용
2. JwtExceptionFilter
JwtAuthFilter의 상위에 존재하며,
JwtAuthFilter에서 발생한  Exception을 Catch해 Response로 반환

JeekLee committed:
feat: Jwt 발급 과정 추가

Jwt 발급 과정 적용

JeekLee committed:
fix: Jwt 발급과정 적용 및 예외처리 기능 수정

1. Jwt 발급 과정 적용
2. 예외처리를 위한 Url 설정이 잘못되어있어 재설정

m1naworld reviewed (approved):
고생하셨습니다! 제가 감히 이런 말씀드리기 뭐하지만,,하하 상직님은 정말 좋은 개발자가 되실 것 같아요! 코드에서부터 다른사람을 생각하는 마음이 보입니다! 감사합니다 🫣

JeongO41 reviewed (approved):
우왕 상직님 코드 감히 리뷰라니.....리뷰는 아니고 스터디 하고 갑니당 :)

JeekLee committed:
fix: 인증/인가 예외처리 기능 수정

예외처리 관련 로직 수정
1. WebSercurity Config
 : antMatchers를 사용해 HttpMethod를
기준으로 예외처리
2. WebSercurityConfig
 : response.getMehtod를 통해 HttpMethod를
기준으로 예외처리

JeekLee commented:
엄청 쉬운 방법을 앞에두고 되게 뻘짓 했었습니다ㅠ 리뷰해주셔서 감사합니다!
